### PR TITLE
docs(schema): add new 'Getting Started with YAML' page

### PR DESCRIPTION
**Pull request by hassankhan**

### CODE REVIEW

**Review by keydunov — CHANGES_REQUESTED** [review]
- Let's fix the camel case
- Let's have a section on top explaining the case (js vs yaml)
- Let's remove js examples and make text explanation

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+### <--{"id" : "Syntax"}--> Defining cubes
+
+```yaml
+cubes:
+  - name: Orders
+    sql: SELECT * FROM public.orders
+```
+
+Multi-line SQL queries also work:
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:42]: Example is not multi-line

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+either `CUBE` or another foreign cube, ensure they are wrapped in `{}` and that
+the entire statement is wrapped in `""`:
+
+```yaml
+cubes:
+  - name: Orders
+    sql: SELECT * FROM public.orders
+    joins:
+      - name: Users
+        sql: '{CUBE}.user_id = {Users.id}'
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:115]: Let's use `"` always

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        relationship: belongs_to
+```
+
+### <--{"id" : "Syntax"}--> Defining pre-aggregations
+
+The `preAggregations` property is an array of [pre-aggregation
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:121]: Let's use snake_case - `pre_aggregations`

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+specify the property directly:
+
+```yaml
+cubes:
+  - name: Orders
+    sql: SELECT * FROM public.orders
+    shown: COMPILE_CONTEXT.securityContext.property
+```
+
+If referencing the security context, ensure that you use camel-case (i.e.
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:159]: @hassankhan why? cc @paveltiunov

**Comment by keydunov** [comment]
@hassankhan Let's remove `drill_members` from all the examples.

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+## Limitations
+
+YAML schemas currently have some limitations compared to JavaScript schemas, the
+main ones being
+
+- Dynamic schemas
+- Code Reusability: No support for
+  [`import`/`export`](https://cube.dev/docs/schema/advanced/export-import)
+- Code Reusability: No support for
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:195]: per @paveltiunov it seems is not longer the limitation

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        type: time
+```
+
+## Limitations
+
+YAML schemas currently have some limitations compared to JavaScript schemas, the
+main ones being
+
+- Dynamic schemas
+- Code Reusability: No support for
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:193]: I don't think we need to list it cc @paveltiunov

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+</WarningBox>
+
+## Ensure deployment is using latest Cube
+
+First, ensure you're using Cube version `0.31.0` or higher. Then create a new
+`.yml` file in the `schema/` folder.
+
+## Syntax
+
+<InfoBox>
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:22]: Should be really infobox?

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+
+1. All schema properties can be written in either camel-case or snake-case i.e.
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:24]: Let's say while you can use both snake case and camel case, we recommend using snake case with YAML.

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+   `countDistinctApprox` can be written as `count_distinct_approx`,
+   `preAggregations` can be written as `pre_aggregations` etc.
+
+2. When referencing either `CUBE` or another foreign cube, ensure the reference
+   is wrapped in `{}` and the entire statement is wrapped in quotes i.e.
+   `'{CUBE}.user_id = {Users.id}'`.
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:30]: Let's use double quotes here. Also let's say that for strings you don't normally need to use quotes, unless you want your numbers to be treated as strings or you use special numbers, including `{}` for interpolation.

File: docs/content/Schema/Getting-Started-with-YAML.mdx (hunk 1 above)
Comment (hassankhan) [reply docs/content/Schema/Getting-Started-with-YAML.mdx:44]: Prettier reformatted it to be on a single line, fixed now

File: docs/content/Schema/Getting-Started-with-YAML.mdx [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+Multi-line SQL queries also work:
+
+```yaml
+cubes:
+  - name: Orders
+    sql:
+      SELECT * FROM public.orders WHERE
+      {FILTER_PARAMS.Orders.created_at.filter('created_at')}
```
Comment (keydunov) [inline docs/content/Schema/Getting-Started-with-YAML.mdx:51]: Will that work? it has `{}`

**Pull request merged**